main.cpp

```diff
@@ -232,265 +291,392 @@ struct Gym
+    return totalPrice;
+}
+
+void Gym::setPersonalTrainer(std::string personalTrainerName, float sessionLength, double sessionTime)
+{
+    personalTrainerName = "Dave";
+    sessionLength = 1.5f;
+    sessionTime = 2;
+
+    std::cout << "You have scheduled a session with: " << personalTrainerName << "\n at: " << sessionTime << "\n for: " << sessionLength << " hours\n";
+}
+
+double Gym::getUserFreq(double numDailyPeople, int numDays)
+{
+    double userFreq = numDailyPeople / numDays;
+    return userFreq;
+}
+
+void Gym::Member::punchPass(Gym::Member, int passesRemaining)
+{
+    Member existingMember;
+    passesLeft -= 1;
+    passesRemaining = passesLeft;
+    std::cout << "you have " << passesRemaining << " passes left!\n";
+}
+
+bool Gym::Member::bringAFriend(bool hasFriendPass, int friendPassType)
+{
+    if (hasFriendPass == true)
+    {
+        passesLeft -= 1;
+    }
+
+    std::cout << "you used " << friendPassType << " and have " << passesLeft << " passes left\n";
+    return hasFriendPass;
+}
+
+int Gym::Member::giveReward(int subsctiptionType, std::string rewardName, int rewardAmount)
+{
+    if (subsctiptionType >= 4; subsctiptionType < 8)
+    {
+        rewardName = "10% discount";
+        rewardAmount = 10;
+    }
+    else if (subsctiptionType >8)
+    {
+        rewardName = "20% discount";
+        rewardAmount = 20;
+    }
+    else
+    {
+        rewardName = "none";
+        rewardAmount = 0;
+    }
+    return rewardAmount;
+}
+
 struct Piano
 {
-    //number of keys
     int totalNumKeys = 88;
-    //type of piano
     std::string pianoType = "Grand Piano";
-    //size of piano
     double pianoSize = 20.80;
-    //name of brand
     std::string brandName = "Yamaha";
-    //number of pedals
     int totalNumPedals = 3;
 
-    //play notes
-    int playNote(int numKey, float playTime = 45.0f);
-    //mute notes
+    double playNote(int numKey, double playTime = 45.0);
     int muteNote(bool playOrMute = false, bool stringPadOn = true);
-    //change volume of notes
     int volumeDown(bool quietMode = true, int numPedal = 2);
 };
 
- /*
-Thing 5) Interface
-5 properties:
-    1) parameter knobs (position) (float)
-    2) parameter sliders (position) (float)
-    3) menu buttons (int)
-    4) keys (int)
-    5) mod wheels (position) (double)
-3 things it can do:
-    1) Play notes
-    2) Change Parameters
-    3) Lookup menus
- */
+double Piano::playNote(int numKey, double playTime)
+{
+    std::cout << "what key would you like to play? (insert key number)\n";
+    std::cin >> numKey;
+    std::cout << "how long would you like to play the note? (insert time in ms)\n";
+    std::cin >> playTime;
+    return playTime;
+}
+
+int Piano::muteNote(bool playOrMute, bool stringPadOn)
+{
+    if (playOrMute == false)
+    {
+        return 0;
+    }
+    else
+    {
+        if (stringPadOn == true)
+        {
+            return 50;
+        }
+        else
+        {
+            return 100;
+        }
+    }
+}
```

Comment:
no need for that last `else`
see previous comments about `else { return }`
- [ ] revise